include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -126,7 +125,8 @@ SpatialOperatorBase<dim, Number>::fill_matrix_free_data(
     matrix_free_data.append_mapping_flags(
       Operators::ViscousKernel<dim, Number>::get_mapping_flags(true, true));
 
-  if(param.right_hand_side)
+  // RHSOperator is used to initialize the velocity field for HDIV!
+  if(param.right_hand_side || param.spatial_discretization == SpatialDiscretization::HDIV)
```

Comment:
why is this still necessary? I thought that the initialization of the velocity field is now done via `dealii::VectorTools`?

Reply:
I have not seen places in the code using the right-hand side operator for HDIV. So let's make sure that this part of the code is correct.

Reply:
Yes, I forgot to remove it when I changed back to `VT::interpolate`

Reply:
This was lastly changed here https://github.com/exadg/exadg/pull/233/commits/9b5780da19d7f31d6c490316d1e10f5297ab78bb